components/sensor/daly_hkms_bms.rst

```diff
@@ -0,0 +1,226 @@
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **cells_number** (*Optional*): The number of cells in series in the battery pack.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **max_cell_voltage** (*Optional*): The cell of the battery with the highest voltage.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **max_cell_voltage_number** (*Optional*): The cell number of the battery with the highest voltage.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **min_cell_voltage** (*Optional*): The cell of the battery with the lowest voltage.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **min_cell_voltage_number** (*Optional*): The cell number of the battery with the lowest voltage.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **cell_1_voltage** (*Optional*): The voltage of cell number 1. Cell number can be from 1 to 48.
+  All options from :ref:`Sensor <config-sensor>`.
+
+Text Sensor
+-----------
+
+Text sensor that indicates the status of BMS.
+
+.. code-block:: yaml
+
+    text_sensor:
+      - platform: daly_hkms_bms
+        status:
+          name: "BMS Status"
+
+Configuration variables:
+************************
+
+- **daly_hkms_bms_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the daly_hkms_bms hub if you want
+  to use multiple BMS.
+
+- **status** (*Optional*): The BMS Status (Charging, Discharging, Stationary).
+  All options from :ref:`Text Sensor <config-text_sensor>`.
+
+Binary Sensor
+-------------
+
+Binary sensor that indicates the status of MOS.
+
+.. code-block:: yaml
+
+    binary_sensor:
+      - platform: daly_hkms_bms
+        charging_mos_enabled:
+          name: "Charging MOS"
+        discharging_mos_enabled:
+          name: "Discharging MOS"
+
+Configuration variables:
+************************
+
+- **daly_hkms_bms_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the daly_hkms_bms hub if you want
+  to use multiple BMS.
+
+- **charging_mos_enabled** (*Optional*): The BMS charging MOS status to enable the recharge of the battery.
+  All options from :ref:`Binary Sensor <config-binary_sensor>`.
+
+- **discharging_mos_enabled** (*Optional*): The BMS discharging mos status to enable the load.
+  All options from :ref:`Binary Sensor <config-binary_sensor>`.
+
+- **precharging_mos_enabled** (*Optional*): The BMS precharging mos status to precharge the load's capacitors.
+  All options from :ref:`Binary Sensor <config-binary_sensor>`.
+
+- **balancing_active** (*Optional*): Whether the BMS is currently balancing the cells.
+  All options from :ref:`Binary Sensor <config-binary_sensor>`.
+
+UART Connection
+---------------
+
+Connect BMS RX to ESP TX, BMS TX to ESP RX and BMS GND to ESP GND.
+
+Alternatively, use an RS485 converter and connect A to A/+ and B to B/-.
+Always connect the ground, even when using RS485, to avoid stray currents running through the RS485 transceivers.
```

Comment:
After your refactor, I think my suggestion no longer makes sense. This reverts it:

```suggestion
Always connect the ground when using RS485 to avoid stray currents running through the RS485 transceivers.
```